pkg/mover/auth.go

```diff
@@ -27,3 +28,13 @@ func (repo ContainerRepository) Resolve(resource authn.Resource) (authn.Authenti
 func (repo ContainerRepository) Authorization() (*authn.AuthConfig, error) {
 	return &authn.AuthConfig{Username: repo.Username, Password: repo.Password}, nil
 }
+
+// Define a container images keychain based on the settings provided via the containers struct
+// If useDefaultKeychain is set, use config/docker.json otherwise load the provided creds (if any)
+func getContainersKeychain(c Containers) authn.Keychain {
+	if c.UseDefaultLocalKeychain {
+		return authn.DefaultKeychain
+	}
+
+	return c.ContainerRepository
+}
```

Comment:
uhm so if I understood this correctly, we can either:
- Use the default key chain 
OR
- Use the explicit repository auth with no fallback, or falling back to anonymous access.

Are we sure we won't want to be able to fallback to the default key chain, specially thinking on CLI usage.

Reply:
Thanks Jose.

That's right, effectively now you choose the authentication mechanism, a) local/default keychain or b) explicit provided credentials. There is not interaction between them.

> Are we sure we won't want to be able to fallback to the default key chain, specially thinking on CLI usage.

Just to clarify, as of today the CLI still uses local keychain while in charts-syncer use explicit credentials via the library. I am not sure if whenever the CLI supports explicit credentials a fallback mechanism will make sense. Libraries like for example viper, assume that what you explicitly set via env variables/flags overrides what's set in config files, so I am seeing this change in that direction.

Falling back can be very confusing and lead to "it works in my computer" kind of a thing. That's what happened to me in charts-syncer [PR](https://github.com/bitnami-labs/charts-syncer/pull/134) where I was passing credentials but it was falling back to local which was fine.

In any case, if you have in mind any CLI that does a good use of a similar fallback mechanism let me know. 

WDYT?

Reply:
Makes sense.